packages/terra-paginator/src/Paginator.jsx

```diff
@@ -0,0 +1,229 @@
+    const previousPageIndex = selectedPage === 1 ? 1 : selectedPage - 1;
+    const nextPageIndex = selectedPage === totalPages ? totalPages : selectedPage + 1;
+
+    const fullView = (
+      <div className={cx(['paginator', !this.hasNavContext() && 'pageless'])}>
+        {
+          this.hasNavContext() && <a
+            className={cx(['nav-link', 'left-controls', selectedPage === 1 && 'is-disabled'])}
+            tabIndex="0"
+            onClick={this.handlePageChange(1)}
+            onKeyDown={this.handleOnKeyDown(1)}
+          >
+            First
+          </a>
+        }
+        <a
+          className={cx(['nav-link', 'left-controls', 'previous', selectedPage === 1 && 'is-disabled'])}
+          tabIndex="0"
+          onClick={this.handlePageChange(previousPageIndex)}
+          onKeyDown={this.handleOnKeyDown(previousPageIndex)}
+        >
+          <span className={cx('icon')} />Previous
+        </a>
+        {this.hasNavContext() && this.buildPageButtons(totalPages, this.handlePageChange)}
+        <a
+          className={cx(['nav-link', 'right-controls', 'next', selectedPage === totalPages && 'is-disabled'])}
+          tabIndex="0"
+          onClick={this.handlePageChange(nextPageIndex)}
+          onKeyDown={this.handleOnKeyDown(nextPageIndex)}
+        >
+          Next<span className={cx('icon')} />
+        </a>
+        {
+          this.hasNavContext() && <a
+            className={cx(['nav-link', 'right-controls', selectedPage === totalPages && 'is-disabled'])}
+            tabIndex="0"
+            onClick={this.handlePageChange(totalPages)}
+            onKeyDown={this.handleOnKeyDown(totalPages)}
+          >
+            Last
+          </a>
+        }
+      </div>
+    );
+
+    return fullView;
+  }
+
+  reducedPaginator() {
+    const totalPages = calculatePages(this.props.totalCount, this.props.itemCountPerPage);
+    const { selectedPage } = this.state;
+
+    const reducedView = (
+      <div className={cx(['paginator', !this.hasNavContext() && 'pageless'])}>
+        {
+          this.hasNavContext() && <a
+            className={cx(['nav-link', 'left-controls', selectedPage === 1 && 'is-disabled'])}
+            tabIndex="0"
+            onClick={this.handlePageChange(1)}
+          >
+            First
+          </a>
+        }
+        <a
+          className={cx(['nav-link', 'left-controls', 'previous', selectedPage === 1 && 'is-disabled'])}
+          tabIndex="0"
+          onClick={this.handlePageChange(selectedPage === 1 ? 1 : selectedPage - 1)}
+        >
+          <span className={cx('icon')} />
+        </a>
+        {this.hasNavContext() && `Page ${selectedPage}`}
+        <a
+          className={cx(['nav-link', 'right-controls', 'next', selectedPage === totalPages && 'is-disabled'])}
+          tabIndex="0"
+          onClick={this.handlePageChange(selectedPage === totalPages ? totalPages : selectedPage + 1)}
+        >
+          <span className={cx('icon')} />
```

Comment:
Will need to add additional icon-only class to remove extra margin for when there is no text:
`<span className={cx(['icon', 'icon-only'])} />`

Reply:
https://github.com/cerner/terra-core/pull/1445/commits/37187bdacdbd135aef8fab5bf1ca8b24cbbc2bf9